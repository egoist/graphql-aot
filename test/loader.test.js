const path = require('path')
const { promisify } = require('util')
const webpack = require('webpack')
const { makeExecutableSchema } = require('graphql-tools')

jest.setTimeout(10000)

const loader = require.resolve('../lib/loader')

test('main', async () => {
  const MFS = webpack.MemoryOutputFileSystem
  const mfs = new MFS()
  const compiler = webpack({
    entry: path.join(__dirname, 'fixtures/loader-only', 'index.js'),
    mode: 'development',
    devtool: false,
    output: {
      path: '/tmp/'
    },
    module: {
      rules: [
        {
          test: /\.gql$/,
          type: 'json',
          loader,
          options: {
            defaultClientOptions: {
              uri: 'https://api.graph.cool/simple/v1/cjc32dadu12pf0184otnrztnq'
            }
          }
        }
      ]
    }
  })
  compiler.outputFileSystem = mfs

  const stats = await promisify(compiler.run.bind(compiler))()

  if (stats.hasErrors()) {
    throw new Error(stats.toString('errors-only'))
  }

  expect(mfs.readFileSync('/tmp/main.js', 'utf8')).toMatchSnapshot()
})

test('use schema', async () => {
  const MFS = webpack.MemoryOutputFileSystem
  const mfs = new MFS()
  const compiler = webpack({
    entry: path.join(__dirname, 'fixtures/use-schema', 'index.js'),
    mode: 'development',
    devtool: false,
    output: {
      path: '/tmp/'
    },
    module: {
      rules: [
        {
          test: /\.gql$/,
          type: 'json',
          loader,
          options: {
            useSchema: {
              schema: makeExecutableSchema({
                typeDefs: `type Query { hello(name: String): String }`,
                resolvers: {
                  Query: {
                    hello: (root, args, context) => {
                      return `hello ${args.name} ${context.type}`
                    }
                  }
                }
              }),
              getContext() {
                return {
                  type: 'hmm'
                }
              }
            }
          }
        }
      ]
    }
  })
  compiler.outputFileSystem = mfs

  const stats = await promisify(compiler.run.bind(compiler))()

  if (stats.hasErrors()) {
    throw new Error(stats.toString('errors-only'))
  }

  expect(mfs.readFileSync('/tmp/main.js', 'utf8')).toMatchSnapshot()
})
