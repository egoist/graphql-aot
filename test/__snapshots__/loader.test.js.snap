// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, {
/******/ 				configurable: false,
/******/ 				enumerable: true,
/******/ 				get: getter
/******/ 			});
/******/ 		}
/******/ 	};
/******/
/******/ 	// define __esModule on exports
/******/ 	__webpack_require__.r = function(exports) {
/******/ 		Object.defineProperty(exports, '__esModule', { value: true });
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = \\"./test/fixtures/loader-only/index.js\\");
/******/ })
/************************************************************************/
/******/ ({

/***/ \\"./test/fixtures/loader-only/foo.gql\\":
/*!*******************************************!*\\\\
  !*** ./test/fixtures/loader-only/foo.gql ***!
  \\\\*******************************************/
/*! exports provided: allFiles, default */
/***/ (function(module) {

module.exports = {\\"allFiles\\":[]};

/***/ }),

/***/ \\"./test/fixtures/loader-only/index.js\\":
/*!********************************************!*\\\\
  !*** ./test/fixtures/loader-only/index.js ***!
  \\\\********************************************/
/*! no exports provided */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);
/* harmony import */ var _foo_gql__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! ./foo.gql */ \\"./test/fixtures/loader-only/foo.gql\\");
var _foo_gql__WEBPACK_IMPORTED_MODULE_0___namespace = /*#__PURE__*/Object.assign({}, _foo_gql__WEBPACK_IMPORTED_MODULE_0__, {\\"default\\": _foo_gql__WEBPACK_IMPORTED_MODULE_0__});



/***/ })

/******/ });"
`;

exports[`use schema 1`] = `
"/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, {
/******/ 				configurable: false,
/******/ 				enumerable: true,
/******/ 				get: getter
/******/ 			});
/******/ 		}
/******/ 	};
/******/
/******/ 	// define __esModule on exports
/******/ 	__webpack_require__.r = function(exports) {
/******/ 		Object.defineProperty(exports, '__esModule', { value: true });
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = \\"./test/fixtures/use-schema/index.js\\");
/******/ })
/************************************************************************/
/******/ ({

/***/ \\"./test/fixtures/use-schema/index.js\\":
/*!*******************************************!*\\\\
  !*** ./test/fixtures/use-schema/index.js ***!
  \\\\*******************************************/
/*! no exports provided */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);
/* harmony import */ var _query_gql__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! ./query.gql */ \\"./test/fixtures/use-schema/query.gql\\");
var _query_gql__WEBPACK_IMPORTED_MODULE_0___namespace = /*#__PURE__*/Object.assign({}, _query_gql__WEBPACK_IMPORTED_MODULE_0__, {\\"default\\": _query_gql__WEBPACK_IMPORTED_MODULE_0__});



/***/ }),

/***/ \\"./test/fixtures/use-schema/query.gql\\":
/*!********************************************!*\\\\
  !*** ./test/fixtures/use-schema/query.gql ***!
  \\\\********************************************/
/*! exports provided: hello, default */
/***/ (function(module) {

module.exports = {\\"hello\\":\\"hello egoist hmm\\"};

/***/ })

/******/ });"
`;
